### DISCUSSION TITLE
Tabular backend with DmassUmass input for CFD applications

### DISCUSSION THREAD

**Discussion by GiuSirianni** [CoolProp/CoolProp#2249] [Q&A] [answered]
Hello,

I am using CoolProp in a couple of different CFD codes of mine and its been a great companion up to now. 

Recently, i have started looking into look-up tables for computational time comparison reasons. As many of you might know, CFD applications usually use _density_ and _internal energy_ to compute the thermodynamic state, but the tabular backend of CoolProp does not allow for these inputs. 

I have been searching around for a reason on why this might be, but am unable to find one since many CFD codes are using look-up tables without a problem (usually in the non-ideal gas-dynamics field).

I saw there is a pull request january 2022 that never got accepted. Is there a reason for it?

Thank you in advance.

**Comment by ibell** [comment, marked as answer]
CoolProp does not include UD inputs because UD create an domain for
tabulation that is difficult to turn into a rectangular domain, which is
required for the tabular methods in their current form.

**Reply by GiuSirianni** [reply]
Understood. I have been using with reliable results the UD tabular implementation from @joel-martin (https://github.com/CoolProp/CoolProp/pull/2077). I didn't test it thoroughly, but it has been working for me.

**Comment by ibell** [comment]
Glad to hear it. I have not merged that PR because it only offers partial coverage of the phase diagram, while I think we need a more robust method be merged in CoolProp
